pkg/envoy/accesslog_server.go

```diff
@@ -46,9 +46,13 @@ func StartAccessLogServer(stateDir string, xdsServer *XDSServer) {
 
 	// Make the socket accessible by non-root Envoy proxies, e.g. running in
 	// sidecar containers.
-	if err = os.Chmod(accessLogPath, 0777); err != nil {
+	if err = os.Chmod(accessLogPath, 0660); err != nil {
 		log.WithError(err).Fatalf("Envoy: Failed to change mode of access log listen socket at %s", accessLogPath)
 	}
+	// Change the group to 1337 which is used by Istio sidecars
+	if err = os.Chown(accessLogPath, -1, 1337); err != nil {
+		log.WithError(err).Fatalf("Envoy: Failed to change group of access log listen socket at %s", accessLogPath)
+	}
```

Comment:
is access log only used by istio ? as per below page, it seems to be for any generic go extension as well ?

https://docs.cilium.io/en/latest/concepts/security/proxy/envoy/#id1

Reply:
These sockets are used via the `cilium-envoy` host proxy as well, which is spawned by Cilium agent, so the r/w permissions for the owner allow access from `cilium-proxy`, including Go extensions loaded by it.

Reply:
I fixed the comments:
```
	// Make the socket accessible by owner and group only. Group access is needed for Istio
	// sidecar proxies.
	if err = os.Chmod(accessLogPath, 0660); err != nil {
		log.WithError(err).Fatalf("Envoy: Failed to change mode of access log listen socket at %s", accessLogPath)
	}
	// Change the group to ProxyGID allowing access from any process from that group.
	if err = os.Chown(accessLogPath, -1, option.Config.ProxyGID); err != nil {
		log.WithError(err).Fatalf("Envoy: Failed to change group of access log listen socket at %s", accessLogPath)
	}
```
& the same for the xDS socket.